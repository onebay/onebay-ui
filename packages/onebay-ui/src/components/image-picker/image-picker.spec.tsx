import { mount } from '@vue/test-utils'
import ImagePicker from './index'

const factory = (values = {}, slots = { default: [] }) => {
  return mount(ImagePicker, {
    slots,
    props: { ...values },
  })
}

const files = [
  {
    url: 'https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg',
    id: '2121',
  },
  {
    url: 'https://zos.alipayobjects.com/rmsportal/hqQWgTXdrlmVVYi.jpeg',
    id: '2122',
  },
  {
    url: 'https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg',
    id: '2121',
  },
  {
    url: 'https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg',
    id: '2121',
  },
  {
    url: 'https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg',
    id: '2121',
  },
]
describe('ImagePicker Snap', () => {

  it('render initial ImagePicker', () => {
    const wrapper = factory()
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render ImagePicker -- props className', () => {
    const wrapper = factory({ className: 'test' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render ImagePicker -- props style', () => {
    const wrapper = factory({ style: 'color:red;' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render ImagePicker -- props files', () => {
    const wrapper = factory({ files: files })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render ImagePicker -- props length', () => {
    const wrapper = factory({ files: files, length: 5 })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render ImagePicker -- props mode', () => {
    const wrapper = factory({ files: files, mode: 'top' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render ImagePicker -- props showAddBtn', () => {
    const wrapper = factory({ files: files, showAddBtn: false })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render ImagePicker -- props multiple', () => {
    const wrapper = factory({ files: files, multiple: true })
    expect(wrapper.element).toMatchSnapshot()
  })
})

describe('ImagePicker Event', () => {
  it('ImagePicker onImageClick', async () => {
    const onImageClick = jest.fn()
    const wrapper = factory({ files: files })
    wrapper.find('.ob-image-picker__preview-img').trigger('click')
    await wrapper.setProps({ onImageClick })
    wrapper.find('.ob-image-picker__preview-img').trigger('click')
    expect(onImageClick).toBeCalled()
  })
  it('ImagePicker remove image', () => {
    window.URL.revokeObjectURL = () => { }
    const wrapper = factory({ files: files })
    wrapper.find('.ob-image-picker__remove-btn').trigger('click')
  })
  it('ImagePicker choose File', () => {
    const wrapper = factory({ files: files, multiple: true })
    wrapper.find('.ob-image-picker__choose-btn').trigger('click')
  })
})