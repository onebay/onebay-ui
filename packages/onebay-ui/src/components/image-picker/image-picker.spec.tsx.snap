// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImagePicker Snap render ImagePicker -- props className 1`] = `
<div
  class="ob-image-picker test"
>
  
  <div
    class="ob-image-picker__flex-box"
  >
    
    <div
      class="ob-image-picker__flex-item"
    >
      <div
        class="ob-image-picker__item ob-image-picker__choose-btn"
      >
        <div
          class="add-bar"
        />
        <div
          class="add-bar"
        />
      </div>
    </div>
    <div
      class="ob-image-picker__flex-item"
    >
      <!---->
    </div>
    <div
      class="ob-image-picker__flex-item"
    >
      <!---->
    </div>
    <div
      class="ob-image-picker__flex-item"
    >
      <!---->
    </div>
    
  </div>
  
</div>
`;

exports[`ImagePicker Snap render ImagePicker -- props customStyle 1`] = `
<div
  class="ob-image-picker"
  style="color: red;"
>
  
  <div
    class="ob-image-picker__flex-box"
  >
    
    <div
      class="ob-image-picker__flex-item"
    >
      <div
        class="ob-image-picker__item ob-image-picker__choose-btn"
      >
        <div
          class="add-bar"
        />
        <div
          class="add-bar"
        />
      </div>
    </div>
    <div
      class="ob-image-picker__flex-item"
    >
      <!---->
    </div>
    <div
      class="ob-image-picker__flex-item"
    >
      <!---->
    </div>
    <div
      class="ob-image-picker__flex-item"
    >
      <!---->
    </div>
    
  </div>
  
</div>
`;

exports[`ImagePicker Snap render ImagePicker -- props files 1`] = `
<div
  class="ob-image-picker"
>
  
  <div
    class="ob-image-picker__flex-box"
  >
    
    <div
      class="ob-image-picker__flex-item"
    >
      <div
        class="ob-image-picker__item"
      >
        <div
          class="ob-image-picker__remove-btn"
        />
        <img
          class="ob-image-picker__preview-img"
          src="https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg"
        />
      </div>
    </div>
    <div
      class="ob-image-picker__flex-item"
    >
      <div
        class="ob-image-picker__item"
      >
        <div
          class="ob-image-picker__remove-btn"
        />
        <img
          class="ob-image-picker__preview-img"
          src="https://zos.alipayobjects.com/rmsportal/hqQWgTXdrlmVVYi.jpeg"
        />
      </div>
    </div>
    <div
      class="ob-image-picker__flex-item"
    >
      <div
        class="ob-image-picker__item"
      >
        <div
          class="ob-image-picker__remove-btn"
        />
        <img
          class="ob-image-picker__preview-img"
          src="https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg"
        />
      </div>
    </div>
    <div
      class="ob-image-picker__flex-item"
    >
      <div
        class="ob-image-picker__item"
      >
        <div
          class="ob-image-picker__remove-btn"
        />
        <img
          class="ob-image-picker__preview-img"
          src="https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg"
        />
      </div>
    </div>
    
  </div>
  <div
    class="ob-image-picker__flex-box"
  >
    
    <div
      class="ob-image-picker__flex-item"
    >
      <div
        class="ob-image-picker__item"
      >
        <div
          class="ob-image-picker__remove-btn"
        />
        <img
          class="ob-image-picker__preview-img"
          src="https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg"
        />
      </div>
    </div>
    <div
      class="ob-image-picker__flex-item"
    >
      <div
        class="ob-image-picker__item ob-image-picker__choose-btn"
      >
        <div
          class="add-bar"
        />
        <div
          class="add-bar"
        />
      </div>
    </div>
    <div
      class="ob-image-picker__flex-item"
    >
      <!---->
    </div>
    <div
      class="ob-image-picker__flex-item"
    >
      <!---->
    </div>
    
  </div>
  
</div>
`;

exports[`ImagePicker Snap render ImagePicker -- props length 1`] = `
<div
  class="ob-image-picker"
>
  
  <div
    class="ob-image-picker__flex-box"
  >
    
    <div
      class="ob-image-picker__flex-item"
    >
      <div
        class="ob-image-picker__item"
      >
        <div
          class="ob-image-picker__remove-btn"
        />
        <img
          class="ob-image-picker__preview-img"
          src="https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg"
        />
      </div>
    </div>
    <div
      class="ob-image-picker__flex-item"
    >
      <div
        class="ob-image-picker__item"
      >
        <div
          class="ob-image-picker__remove-btn"
        />
        <img
          class="ob-image-picker__preview-img"
          src="https://zos.alipayobjects.com/rmsportal/hqQWgTXdrlmVVYi.jpeg"
        />
      </div>
    </div>
    <div
      class="ob-image-picker__flex-item"
    >
      <div
        class="ob-image-picker__item"
      >
        <div
          class="ob-image-picker__remove-btn"
        />
        <img
          class="ob-image-picker__preview-img"
          src="https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg"
        />
      </div>
    </div>
    <div
      class="ob-image-picker__flex-item"
    >
      <div
        class="ob-image-picker__item"
      >
        <div
          class="ob-image-picker__remove-btn"
        />
        <img
          class="ob-image-picker__preview-img"
          src="https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg"
        />
      </div>
    </div>
    <div
      class="ob-image-picker__flex-item"
    >
      <div
        class="ob-image-picker__item"
      >
        <div
          class="ob-image-picker__remove-btn"
        />
        <img
          class="ob-image-picker__preview-img"
          src="https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg"
        />
      </div>
    </div>
    
  </div>
  <div
    class="ob-image-picker__flex-box"
  >
    
    <div
      class="ob-image-picker__flex-item"
    >
      <div
        class="ob-image-picker__item ob-image-picker__choose-btn"
      >
        <div
          class="add-bar"
        />
        <div
          class="add-bar"
        />
      </div>
    </div>
    <div
      class="ob-image-picker__flex-item"
    >
      <!---->
    </div>
    <div
      class="ob-image-picker__flex-item"
    >
      <!---->
    </div>
    <div
      class="ob-image-picker__flex-item"
    >
      <!---->
    </div>
    <div
      class="ob-image-picker__flex-item"
    >
      <!---->
    </div>
    
  </div>
  
</div>
`;

exports[`ImagePicker Snap render ImagePicker -- props mode 1`] = `
<div
  class="ob-image-picker"
  mode="top"
>
  
  <div
    class="ob-image-picker__flex-box"
  >
    
    <div
      class="ob-image-picker__flex-item"
    >
      <div
        class="ob-image-picker__item"
      >
        <div
          class="ob-image-picker__remove-btn"
        />
        <img
          class="ob-image-picker__preview-img"
          src="https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg"
        />
      </div>
    </div>
    <div
      class="ob-image-picker__flex-item"
    >
      <div
        class="ob-image-picker__item"
      >
        <div
          class="ob-image-picker__remove-btn"
        />
        <img
          class="ob-image-picker__preview-img"
          src="https://zos.alipayobjects.com/rmsportal/hqQWgTXdrlmVVYi.jpeg"
        />
      </div>
    </div>
    <div
      class="ob-image-picker__flex-item"
    >
      <div
        class="ob-image-picker__item"
      >
        <div
          class="ob-image-picker__remove-btn"
        />
        <img
          class="ob-image-picker__preview-img"
          src="https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg"
        />
      </div>
    </div>
    <div
      class="ob-image-picker__flex-item"
    >
      <div
        class="ob-image-picker__item"
      >
        <div
          class="ob-image-picker__remove-btn"
        />
        <img
          class="ob-image-picker__preview-img"
          src="https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg"
        />
      </div>
    </div>
    
  </div>
  <div
    class="ob-image-picker__flex-box"
  >
    
    <div
      class="ob-image-picker__flex-item"
    >
      <div
        class="ob-image-picker__item"
      >
        <div
          class="ob-image-picker__remove-btn"
        />
        <img
          class="ob-image-picker__preview-img"
          src="https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg"
        />
      </div>
    </div>
    <div
      class="ob-image-picker__flex-item"
    >
      <div
        class="ob-image-picker__item ob-image-picker__choose-btn"
      >
        <div
          class="add-bar"
        />
        <div
          class="add-bar"
        />
      </div>
    </div>
    <div
      class="ob-image-picker__flex-item"
    >
      <!---->
    </div>
    <div
      class="ob-image-picker__flex-item"
    >
      <!---->
    </div>
    
  </div>
  
</div>
`;

exports[`ImagePicker Snap render ImagePicker -- props multiple 1`] = `
<div
  class="ob-image-picker"
>
  
  <div
    class="ob-image-picker__flex-box"
  >
    
    <div
      class="ob-image-picker__flex-item"
    >
      <div
        class="ob-image-picker__item"
      >
        <div
          class="ob-image-picker__remove-btn"
        />
        <img
          class="ob-image-picker__preview-img"
          src="https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg"
        />
      </div>
    </div>
    <div
      class="ob-image-picker__flex-item"
    >
      <div
        class="ob-image-picker__item"
      >
        <div
          class="ob-image-picker__remove-btn"
        />
        <img
          class="ob-image-picker__preview-img"
          src="https://zos.alipayobjects.com/rmsportal/hqQWgTXdrlmVVYi.jpeg"
        />
      </div>
    </div>
    <div
      class="ob-image-picker__flex-item"
    >
      <div
        class="ob-image-picker__item"
      >
        <div
          class="ob-image-picker__remove-btn"
        />
        <img
          class="ob-image-picker__preview-img"
          src="https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg"
        />
      </div>
    </div>
    <div
      class="ob-image-picker__flex-item"
    >
      <div
        class="ob-image-picker__item"
      >
        <div
          class="ob-image-picker__remove-btn"
        />
        <img
          class="ob-image-picker__preview-img"
          src="https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg"
        />
      </div>
    </div>
    
  </div>
  <div
    class="ob-image-picker__flex-box"
  >
    
    <div
      class="ob-image-picker__flex-item"
    >
      <div
        class="ob-image-picker__item"
      >
        <div
          class="ob-image-picker__remove-btn"
        />
        <img
          class="ob-image-picker__preview-img"
          src="https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg"
        />
      </div>
    </div>
    <div
      class="ob-image-picker__flex-item"
    >
      <div
        class="ob-image-picker__item ob-image-picker__choose-btn"
      >
        <div
          class="add-bar"
        />
        <div
          class="add-bar"
        />
      </div>
    </div>
    <div
      class="ob-image-picker__flex-item"
    >
      <!---->
    </div>
    <div
      class="ob-image-picker__flex-item"
    >
      <!---->
    </div>
    
  </div>
  
</div>
`;

exports[`ImagePicker Snap render ImagePicker -- props showAddBtn 1`] = `
<div
  class="ob-image-picker"
>
  
  <div
    class="ob-image-picker__flex-box"
  >
    
    <div
      class="ob-image-picker__flex-item"
    >
      <div
        class="ob-image-picker__item"
      >
        <div
          class="ob-image-picker__remove-btn"
        />
        <img
          class="ob-image-picker__preview-img"
          src="https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg"
        />
      </div>
    </div>
    <div
      class="ob-image-picker__flex-item"
    >
      <div
        class="ob-image-picker__item"
      >
        <div
          class="ob-image-picker__remove-btn"
        />
        <img
          class="ob-image-picker__preview-img"
          src="https://zos.alipayobjects.com/rmsportal/hqQWgTXdrlmVVYi.jpeg"
        />
      </div>
    </div>
    <div
      class="ob-image-picker__flex-item"
    >
      <div
        class="ob-image-picker__item"
      >
        <div
          class="ob-image-picker__remove-btn"
        />
        <img
          class="ob-image-picker__preview-img"
          src="https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg"
        />
      </div>
    </div>
    <div
      class="ob-image-picker__flex-item"
    >
      <div
        class="ob-image-picker__item"
      >
        <div
          class="ob-image-picker__remove-btn"
        />
        <img
          class="ob-image-picker__preview-img"
          src="https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg"
        />
      </div>
    </div>
    
  </div>
  <div
    class="ob-image-picker__flex-box"
  >
    
    <div
      class="ob-image-picker__flex-item"
    >
      <div
        class="ob-image-picker__item"
      >
        <div
          class="ob-image-picker__remove-btn"
        />
        <img
          class="ob-image-picker__preview-img"
          src="https://zos.alipayobjects.com/rmsportal/PZUUCKTRIHWiZSY.jpeg"
        />
      </div>
    </div>
    <div
      class="ob-image-picker__flex-item"
    >
      <!---->
    </div>
    <div
      class="ob-image-picker__flex-item"
    >
      <!---->
    </div>
    <div
      class="ob-image-picker__flex-item"
    >
      <!---->
    </div>
    
  </div>
  
</div>
`;

exports[`ImagePicker Snap render initial ImagePicker 1`] = `
<div
  class="ob-image-picker"
>
  
  <div
    class="ob-image-picker__flex-box"
  >
    
    <div
      class="ob-image-picker__flex-item"
    >
      <div
        class="ob-image-picker__item ob-image-picker__choose-btn"
      >
        <div
          class="add-bar"
        />
        <div
          class="add-bar"
        />
      </div>
    </div>
    <div
      class="ob-image-picker__flex-item"
    >
      <!---->
    </div>
    <div
      class="ob-image-picker__flex-item"
    >
      <!---->
    </div>
    <div
      class="ob-image-picker__flex-item"
    >
      <!---->
    </div>
    
  </div>
  
</div>
`;
